server/src/main/java/io/unitycatalog/server/service/SchemaService.java

```diff
@@ -1,47 +1,152 @@
 package io.unitycatalog.server.service;
 
+import static io.unitycatalog.server.model.ResourceType.CATALOG;
+import static io.unitycatalog.server.model.ResourceType.METASTORE;
+import static io.unitycatalog.server.model.ResourceType.SCHEMA;
+
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.server.annotation.*;
+import io.unitycatalog.server.auth.UnityCatalogAuthorizer;
+import io.unitycatalog.server.auth.annotation.AuthorizeExpression;
+import io.unitycatalog.server.auth.annotation.AuthorizeKey;
+import io.unitycatalog.server.auth.decorator.UnityAccessEvaluator;
+import io.unitycatalog.server.auth.decorator.UnityAccessUtil;
 import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.model.CatalogInfo;
 import io.unitycatalog.server.model.CreateSchema;
+import io.unitycatalog.server.model.ListSchemasResponse;
+import io.unitycatalog.server.model.Privilege;
+import io.unitycatalog.server.model.SchemaInfo;
 import io.unitycatalog.server.model.UpdateSchema;
+import io.unitycatalog.server.persist.CatalogRepository;
+import io.unitycatalog.server.persist.MetastoreRepository;
 import io.unitycatalog.server.persist.SchemaRepository;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
+import lombok.SneakyThrows;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class SchemaService {
   private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
+  private final UnityCatalogAuthorizer authorizer;
+  private final UnityAccessEvaluator evaluator;
 
-  public SchemaService() {}
+  @SneakyThrows
+  public SchemaService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createSchema(CreateSchema createSchema) {
-    return HttpResponse.ofJson(SCHEMA_REPOSITORY.createSchema(createSchema));
+  @AuthorizeExpression("""
+      #authorize(#principal, #metastore, METASTORE_ADMIN) ||
+      #authorize(#principal, #catalog, CREATE_SCHEMA)
+      """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse createSchema(
+      @AuthorizeKey(value = CATALOG, key = "catalog_name") CreateSchema createSchema) {
+    SchemaInfo schemaInfo = SCHEMA_REPOSITORY.createSchema(createSchema);
+    createAuthorizations(schemaInfo);
```

Comment:
could we unify the name for this across service i.e. initializeAuthorizations/createAuthorizations

Reply:
it would have been great if there were a base class that defined child and parent id. i suppose I could just have a method that accepts the ids specifically instead of entities.